File: app/src/main/java/knf/kuma/tv/details/DetailsDescriptionPresenter.java
package knf.kuma.tv.details;

import android.support.annotation.ColorInt;
import android.support.v17.leanback.widget.AbstractDetailsDescriptionPresenter;

import knf.kuma.pojos.AnimeObject;

public class DetailsDescriptionPresenter
        extends AbstractDetailsDescriptionPresenter {
    @ColorInt
    private int titleColor;
    @ColorInt
    private int bodyColor;

    DetailsDescriptionPresenter(int titleColor, int bodyColor) {
        this.titleColor = titleColor;
        this.bodyColor = bodyColor;
    }

    DetailsDescriptionPresenter() {
        this.titleColor = 0;
        this.bodyColor = 0;
    }

    @Override
    protected void onBindDescription(AbstractDetailsDescriptionPresenter.ViewHolder viewHolder, Object itemData) {
        AnimeObject animeObject = (AnimeObject) itemData;
        viewHolder.getTitle().setText(animeObject.name);
        viewHolder.getSubtitle().setText(animeObject.getGenresString());
        viewHolder.getBody().setText(animeObject.description);
        if (titleColor != 0)
            viewHolder.getTitle().setTextColor(titleColor);
        if (bodyColor != 0) {
            viewHolder.getSubtitle().setTextColor(bodyColor);
            viewHolder.getBody().setTextColor(bodyColor);
        }
    }
}

File: app/src/main/java/knf/kuma/tv/details/DetailsDescriptionPresenter.kt
package knf.kuma.tv.details

import androidx.annotation.ColorInt
import androidx.leanback.widget.AbstractDetailsDescriptionPresenter
import knf.kuma.pojos.av1.DirectoryAV1

class DetailsDescriptionPresenter : AbstractDetailsDescriptionPresenter {
    @ColorInt
    private var titleColor: Int = 0
    @ColorInt
    private var bodyColor: Int = 0

    internal constructor(titleColor: Int, bodyColor: Int) {
        this.titleColor = titleColor
        this.bodyColor = bodyColor
    }

    internal constructor() {
        this.titleColor = 0
        this.bodyColor = 0
    }

    override fun onBindDescription(viewHolder: ViewHolder, itemData: Any) {
        val animeObject = itemData as DirectoryAV1
        viewHolder.title.text = animeObject.name
        viewHolder.subtitle.text = animeObject.genres.joinToString { it.name }.trim()
        viewHolder.body.text = animeObject.description
        if (titleColor != 0)
            viewHolder.title.setTextColor(titleColor)
        if (bodyColor != 0) {
            viewHolder.subtitle.setTextColor(bodyColor)
            viewHolder.body.setTextColor(bodyColor)
        }
    }
}
